### PR TITLE
treat http 204 as ok status

### PR DESCRIPTION
**Pull request by tuanng-cognite**

### CODE REVIEW

File: cognite/client/_version.py [hunk 1]
```diff
@@ -1,2 +1,2 @@
-__version__ = "2.42.0"
-__api_subversion__ = "V20220125"
+__version__ = "2.43.0"
+__api_subversion__ = "V20220323"
```
Comment (vincent-cognite) [inline cognite/client/_version.py:2]: this one should remain untouched (I don't know when it should be changed, it is undocumented)

Comment (tuanng-cognite) [reply cognite/client/_version.py:2]: Nice to know. We definitely need to document this somewhere

Comment (vincent-cognite) [reply cognite/client/_version.py:2]: So please remove the change on `__api_subversion__` and fix the conflicts to get approval @tuanng-cognite :)

Comment (vincent-cognite) [reply cognite/client/_version.py:2]: Documentation has been requested earlier. Hasn't happened yet.

**Review by vincent-cognite — APPROVED** [review]
👍

File: CHANGELOG.md [hunk 2]
```diff
@@ -14,6 +14,10 @@ Changes are grouped as follows
 - `Fixed` for any bug fixes.
 - `Security` in case of vulnerabilities.
 
+## [2.46.0] - 2022-03-29
+### Changed
+- Treat HTTP 204 as ok status
```
Comment (dmivankov) [inline CHANGELOG.md:19]: which services/endpoints return 204? That'd be more changelog-worthy :slightly_smiling_face: could be a bugfix or preparation for new endpoints, don't have context

Comment (vincent-cognite) [reply CHANGELOG.md:19]: raster support in geospatial api (experimental) is returning 204 when the corresponding raster is empty (NULL in db). We considered using 404 (or 400) too for the same request but it felt more logical to go with 204 (No content)

Comment (dmivankov) [reply CHANGELOG.md:19]: then suggest putting something like "fixed retrieval of empty raster in experimental geospatial api"

**Review by dmivankov — APPROVED** [review]
Looks safe code-wise, but consider expanding changelog entry

File: CHANGELOG.md (hunk 2 above)
Comment (tuanng-cognite) [reply CHANGELOG.md:19]: https://pr-1632.specs.preview.cogniteapp.com/v1.json.html#operation/getRaster
We are returning 204 if there is no data for this endpoint.

This PR is more aligned with https://github.com/cognitedata/service-contracts/blob/master/design_guide/API_design_guide.md#status-codes where 204 should be supported.

**Pull request merged**